docs/performance.md

```diff
@@ -30,17 +30,17 @@ Let's start with an example demonstrating the case:
 
 In order to reduce execution time we can modify the code and perform the merge operation only once:
 ```php
-    /* the inner empty array covers cases when no loops were made */
-    $options = [[]];
+    $options = [];
     foreach ($configurationSources as $source) {
         /* something happens here */
         $options[] = $source->getOptions(); // <- yes, we'll use a little bit more memory
     }
+   
     /* PHP below 5.6 */
-    $options = call_user_func_array('array_merge', $options);
+    $options = call_user_func_array('array_merge', [] + $options); // the empty array covers cases when no loops were made
```

Comment:
@bcremer: this will not work as expected: https://3v4l.org/IjHO1, please update to 'call_user_func_array('array_merge', $options + [[]])' (with a comment that placing '[[]]' as the second operand is important) and I'll merge the change right ahead.

Reply:
@kalessil Updated the PR. Sorry for not checking the change better beforehand.